src/file.cr

```diff
@@ -377,6 +378,15 @@ class File < IO::FileDescriptor
     stat.size
   end
 
+  # Truncates the file to the specified length. Requires a write file descriptor
+  def truncate(length)
```

Comment:
You're interpreting the manpage right, the issue is the buffered IO here. `truncate` should flush first, else you truncate the file and then it's flushed at close -> stuff pending in the write buffer is written.